### PR TITLE
docs: Add some App Router FAQs

### PR DESCRIPTION
**Pull request by leerob**
Based on questions I've seen in issues and discussions.

### CODE REVIEW

File: docs/02-app/index.mdx [hunk 1]
```diff
@@ -3,8 +3,67 @@ title: App Router
...
+
+### How can I redirect from a Server Component?
+
+You can use [`redirect`](/docs/app/api-reference/functions/redirect) to redirect from a page to a relative or absolute URL. [`redirect`](/docs/app/api-reference/functions/redirect) is a temporary (307) redirect, while [`permanentRedirect`](/docs/app/api-reference/functions/permanentRedirect) is a permanent (308) redirect. When these functions are used while streaming UI, they will insert a meta tag to emit the redirect on the client side.
+
+### How can I handle authentication with the App Router?
+
+Here are some common authentication solutions that support the App Router:
+
+- [NextAuth](https://next-auth.js.org/configuration/nextjs#in-app-router)
```
Comment (balazsorban44) [inline docs/02-app/index.mdx:38]:
```suggestion
- [NextAuth.js](https://next-auth.js.org/configuration/nextjs#in-app-router)
```

**Review by mayank1513 — APPROVED** [review]
Great!

**Review by styfle — APPROVED** [review]

**Comment by GabenGar** [comment]
It looks like the redirect functions are hardcoded to `307`/`308` redirects. How do you do `301`/`302`/`303` redirects then?

**Comment by leerob** [comment]
> Why does Next.js use 307 and 308? Traditionally a 302 was used for a temporary redirect, and a 301 for a permanent redirect, but many browsers changed the request method of the redirect to GET, regardless of the original method. For example, if the browser made a request to POST /v1/users which returned status code 302 with location /v2/users, the subsequent request might be GET /v2/users instead of the expected POST /v2/users. Next.js uses the 307 temporary redirect, and 308 permanent redirect status codes to explicitly preserve the request method used.

From the docs 🙏

**Comment by GabenGar** [comment]
Yeah but there might be use cases for when you want to change the method to `GET` and `301`/`302`/`303` are go-to SEO redirects (SEO doesn't operate outside of `GET` requests anyway). I have a suspicion some search engines can choke on `307`/`308` by assuming these are invalid codes.

**Pull request merged**